ee/clickhouse/client.py

```diff
@@ -74,6 +85,18 @@ def async_execute(query, args=None):
         connections_max=100,
     )
 
+    def cache_sync_execute(query, args=None, redis_client=None, ttl=CACHE_TTL):
+        if not redis_client:
+            redis_client = redis.get_client()
+        key = _key_hash(query, args)
+        if redis_client.exists(key):
+            result = pickle.loads(redis_client.get(key))
+            return result
+        else:
+            result = sync_execute(query, args)
+            redis_client.set(key, pickle.dumps(result), ex=ttl)
```

Comment:
Could we use json module instead? Given that the data returned from clickhouse should be pretty standard, I think it would work well here.

https://docs.python.org/3/library/pickle.html has a big red banner saying it's not secure and I can see that being an issue for us here as well, given parts of the data (event names, payloads) will be untrusted. I remember issues around the format breaking across versions or library upgrades.

Reply:
This is a good point. I chose it just because pickle is faster, but yeah I'm totally ok with JSON instead. There is always a chance that it could mess up with JSON but for now I think this is fine.